.github/workflows/publish-sql-schema.yml

```diff
@@ -23,9 +29,30 @@ jobs:
       id: get_version
       run: |
         VERSION=${GITHUB_REF/refs\/tags\//}
-        echo "MINOR_VERSION=$(echo ${VERSION} | awk -F'.' '{if (NF != 3) {exit 1}; printf "%s.%s", $1, $2}')" >> $GITHUB_OUTPUT
+        echo "minor_version=$(echo ${VERSION} | awk -F'.' '{if (NF != 3) {exit 1}; printf "%s.%s", $1, $2}')" >> $GITHUB_OUTPUT
+    - name: Check validity of migrations
+      uses: "./.github/actions/validate-migrations"
+    - name: Check that existing migrations have not changed
+      if: '!inputs.force'
+      run: |
+        TEMP=$(mktemp -d)
+        gcloud alpha storage cp --recursive \
+          "gs://janus-artifacts-sql-schemas/${{ steps.get_version.outputs.minor_version }}/db" \
+          "$TEMP"
+        if ! git diff --no-index --name-status --diff-filter=CDMRTUXB "$TEMP/db" "./db"; then
```

Comment:
Last thing to consider is whether or not we want `--name-status`. IIUC, the output of `git diff` will go to stdout and show up in the action log, and I think it'd be more helpful to show the entire diff instead of just which files changed. Either way is fine with me, though.

Reply:
My thought was as so: if I get dinged by this check and it tells me that "files cannot be modified", my action is simply to `git restore/revert` whatever was changed. Therefore the most salient information is _what_ files were modified rather than the actual differences. It's a minor optimization though, I can see the full diff still being useful.